File: imagepicker/src/main/java/com/esafirm/imagepicker/features/common/BasePresenter.java
package com.esafirm.imagepicker.features.common;

public class BasePresenter<T extends MvpView> {

    private T view;

    public void attachView(T view) {
        this.view = view;
    }

    public T getView() {
        return view;
    }

    public void detachView() {
        view = null;
    }

    protected boolean isViewAttached() {
        return view != null;
    }
}

File: imagepicker/src/main/java/com/esafirm/imagepicker/features/common/BasePresenter.kt
package com.esafirm.imagepicker.features.common

open class BasePresenter<T : MvpView?> {

    var view: T? = null
        private set

    fun attachView(view: T) {
        this.view = view
    }

    fun detachView() {
        view = null
    }

    protected val isViewAttached: Boolean
        get() = view != null
}